tss-esapi/src/abstraction/public.rs

```diff
@@ -0,0 +1,116 @@
+use crate::interface_types::ecc::EccCurve;
+use crate::structures::{Public, RsaExponent};
+use crate::{Error, WrapperErrorKind};
+
+use core::convert::TryFrom;
+use oid::ObjectIdentifier;
+use picky_asn1::bit_string::BitString;
+use picky_asn1::wrapper::{IntegerAsn1, OctetStringAsn1};
+use picky_asn1_x509::{
+    AlgorithmIdentifier, EcParameters, EcPoint, PublicKey, RsaPublicKey, SubjectPublicKeyInfo,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub enum DecodedKey {
```

Comment:
Could you add some doc here to explain what the purpose of the type is?

Reply:
The question is if we actually need the conversion to `RsaPublicKey` and `EcPoint` exposed.
`EcPoint` is not really usable when you don't know the curve.

Reply:
At least for Parsec it helps - that's the format we export public keys in (the other metadata, such as the exact curve, is obtained separately)

Reply:
done